posthog/warehouse/data_load/pipeline.py

```diff
@@ -0,0 +1,130 @@
+import dlt
+from typing import Dict, List
+from django.conf import settings
+from .stripe import stripe_source, ENDPOINTS
+from dataclasses import dataclass
+from posthog.warehouse.models import ExternalDataSource, DataWarehouseTable
+import s3fs
+from asgiref.sync import sync_to_async
+
+
+@dataclass
+class PipelineInputs:
+    dataset_name: str
+    job_type: str
+    team_id: int
+
+
+def create_pipeline(inputs: PipelineInputs):
+    return dlt.pipeline(
+        pipeline_name=f"{inputs.job_type}_pipeline",
+        destination="filesystem",
+        dataset_name=inputs.dataset_name,
+        credentials={
+            "aws_access_key_id": settings.AIRBYTE_BUCKET_KEY,
+            "aws_secret_access_key": settings.AIRBYTE_BUCKET_SECRET,
+        },
+    )
+
+
+@dataclass
+class SourceColumnType:
+    name: str
+    data_type: str
+    nullable: bool
+
+
+@dataclass
+class SourceSchema:
+    resource: str
+    name: str
+    columns: Dict[str, SourceColumnType]
+    write_disposition: str
+
+
+@dataclass
+class StripeJobInputs(PipelineInputs):
+    stripe_secret_key: str
+
+
+PIPELINE_TYPE_SCHEMA_DEFAULT_MAPPING = {ExternalDataSource.Type.STRIPE: ENDPOINTS}
+
+
+# Run pipeline on separate thread. No db clients used
+@sync_to_async(thread_sensitive=False)
```

Comment:
Unclear to me how to run this pipeline within an async function at the moment so converting this to async with thread_sensitive=false seems the best option for initial rollout